Backend/django_mobx_test/restaurant/views.py

```diff
@@ -0,0 +1,182 @@
+# Create your views here.
+from rest_framework import status
+from rest_framework.authtoken.serializers import AuthTokenSerializer
+from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet, ViewSet
+from rest_framework.authtoken.models import Token
+from restaurant.models import Restaurant, Ticket
+from rest_framework.decorators import action
+
+from restaurant.serializers import SignupSerializer, UserSerializer, RestaurantSerializer, RestaurantModelSerializer, \
+    TicketModelSerializer, TicketSerializer
+
+
+class SignupViewSet(ModelViewSet):
+    authentication_classes = ()
+    permission_classes = (AllowAny,)
+    serializer_class = SignupSerializer
+    http_method_names = ['post']
+
+
+class LoginViewSet(ViewSet):
+    """Based on rest_framework.authtoken.views.ObtainAuthToken"""
+    serializer_class = AuthTokenSerializer
+    permission_classes = (AllowAny,)
+
+    def create(self, request):
+        try:
+            print("enter in create method")
+            print(request.data)
+            serializer = AuthTokenSerializer(
+                data=request.data,
+                context={'request': request}
+            )
+            serializer.is_valid(raise_exception=True)
+            user = serializer.validated_data['user']
+            print(user.id)
+            print("after verification")
+
+            token, created = Token.objects.get_or_create(user=user)
+            user_serializer = UserSerializer(user)
+            return Response({'token': token.key, 'user': user_serializer.data})
+        except Exception as e:
+            return Response({"code": e.args[0]}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+class RestaurantViewSet(ViewSet):
+    permission_classes = [IsAuthenticated, ]
+    serializer_class = RestaurantSerializer
+
+    def create(self, request):
+        try:
```

Comment:
You can simply use ModelViewSet, in case you don't have to define all these CURD views